client.py

```diff
@@ -84,30 +84,30 @@ def parse_ServerResponse(self, response):
 if args.verbose:
     print(res1)
 
-session_id = res1.getroot().attrib['SessionID']
+session_id = res1.attrib['SessionID']
 k = res1.find('.//{http://www.w3.org/2000/09/xmldsig#}RSAKeyValue')
 mod = number.bytes_to_long(d64sb(k.find(
   '{http://www.w3.org/2000/09/xmldsig#}Modulus').text))
 exp = number.bytes_to_long(d64sb(k.find(
   '{http://www.w3.org/2000/09/xmldsig#}Exponent').text))
-pubk = RSA.construct({'n': mod, 'e': exp})
+pubk = RSA.construct( (int(mod), int(exp)) )
 pl = res1.find('.//Payload')
 server_nonce = d64sb(pl.find('Nonce').text)
 
 print("Got server nonce and RSA pubkey:\n{}\n{}".format(
     hexlify(server_nonce), pubk.exportKey('PEM').decode()))
 
 # generate and encrypt client nonce
-client_nonce = random.getrandbits(16*8)
+client_nonce = bytearray([random.getrandbits(8) for i in range(16)])
 cipher = PKCS1_OAEP.new(pubk)
-encrypted_client_nonce = cipher.encrypt(client_nonce)
+encrypted_client_nonce = cipher.encrypt(hexlify(client_nonce))
 print("Generated client nonce:\n\tplaintext: {}\n\tencrypted: {}".format(
     hexlify(client_nonce), hexlify(encrypted_client_nonce)))
 
 # send second request
 req2_filled = req2_tmpl.format(
-  session_id=session_id, encrypted_client_nonce=encrypted_client_nonce,
-  server_nonce=server_nonce)
+  session_id=session_id, encrypted_client_nonce=hexlify(encrypted_client_nonce).decode(),
```

Comment:
ditto

Reply:
Hm, I wonder if these should be base64, like the server nonce is sent?

Reply:
Well that's fascinating, the error changed from Malformed Request to just "Abort"

Reply:
http://dpaste.com/0WGQS25#wrap demangled somewhat.  i'll play with fakeclient <-> fakeserver :P

Reply:
Interesting. The exchanges I captured definitely _did not_ have the x509 certificate embedded.

Here's a complete transcript of the one I saved, de-SOAPified: http://dpaste.com/1P44DBX

Reply:
Yes, mine look more like that too.  No x509 embedded.

Reply:
> Yes, mine look more like that too. No x509 embedded.

So where did you get the one you pasted…?